CONTRIBUTING.md

```diff
@@ -123,6 +124,96 @@ from the main (upstream) repository:
     ```
 
 
+## <a name="commit"></a> Commit Message Guidelines
+
+We have very precise rules over how our git commit messages can be formatted.  This leads to **more
+readable messages** that are easy to follow when looking through the **project history**.  But also,
+we use the git commit messages to **generate changelog**.
+
+### Commit Message Format
+Each commit message consists of a **header**, a **body** and a **footer**.  The header has a special
+format that includes a **type**, a **scope** and a **subject**:
+
+```
+<type>(<scope>): <subject>
+<BLANK LINE>
+<body>
+<BLANK LINE>
+<footer>
+```
+
+The **header** is mandatory and the **scope** of the header is optional.
+
+Any line of the commit message cannot be longer 100 characters! This allows the message to be easier
+to read on GitHub as well as in various git tools.
+
+### Revert
+If the commit reverts a previous commit, it should begin with `revert: `, followed by the header of
+the reverted commit. In the body it should say: `This reverts commit <hash>.`, where the hash is
+the SHA of the commit being reverted.
+
+### Type
+Must be one of the following:
```

Comment:
I don't like changelogs which are same as formatted output of `git log`. I believe changelog should contain only info which might come handy for end users - so only feat, fix, perf types.

Reply:
I agree. Angular preset I used will generate changelog only from `feat`, `fix` and `perf` as you mentioned. It will look like this changelog https://github.com/angular/components/blob/master/CHANGELOG.md.